packages/cursorless-engine/src/core/Debug.ts

```diff
@@ -93,27 +93,55 @@ export class Debug {
     }
 
     const cursor = node.tree.walk();
-    this.printCursorLocationInfo(cursor, 0);
-
-    for (let i = 1; i < ancestors.length; ++i) {
-      cursor.gotoFirstChild();
-      while (cursor.currentNode().id !== ancestors[i].id) {
-        if (!cursor.gotoNextSibling()) {
-          return;
-        }
+    this.printCursorLocationInfo(ancestors, cursor, 0);
+  }
+
+  private printCursorLocationInfo(
+    nodes: SyntaxNode[],
+    cursor: TreeCursor,
+    index: number,
+  ) {
+    const field = cursor.currentFieldName();
+    const fieldText = field != null ? `${field}: ` : "";
+    const ident = " ".repeat(index);
+    const nodeIsLast = index === nodes.length - 1;
+    const { nodeIsNamed } = cursor;
+    let text = `${ident}${fieldText}`;
+
+    if (nodeIsNamed) {
+      text += `(${cursor.nodeType}`;
+      if (nodeIsLast) {
+        text += ")";
       }
-      this.printCursorLocationInfo(cursor, i);
+    } else {
+      text += `"${cursor.nodeType}"`;
+    }
+
+    if (nodeIsLast) {
+      text += " @cursor";
```

Comment:
Mixed feelings as well. Is it not obvious that this is where your cursor is due to the indentation? I think this tag might be a bit confusing if you don't expect it, but maybe it's helpful? Maybe `yourCursor` to make it more obvious? Idk. Could go either way

Reply:
I decided to just drop it